starter-code/src/viking.js

```diff
@@ -1,11 +1,59 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+        }
+
+    attack(){
+        return this.strength;
+    }
+
+    receiveDamage(theDamage) {
+        theDamage = this.health - theDamage;
+        return theDamage
+    }
+    
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    attack(){
+        return this.strength;
+    }
+
+    receiveDamage(theDamage){ 
+        theDamage = this.health - theDamage;
+
+        if (this.health > 0) {
+            console.log(`${name} has recieved ${theDamage} points of damage `);
+        }
+        else if (this.health <= 0) {
+            console.log(`${name} has died in combat`);
+        }
+    }
+} 
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    attack() {
+        return this.strength
+    }
+    receiveDamage(theDamage){
+        this.health -= theDamage;
+        if (this.health > 0){
+            console.log(`a saxson has recieved ${theDamage} points of damage `)
+        } else {
+            console.log(`A saxson has died in combat`)
+        }
+    }
+
+}
 
 // War
 class War {}
```

Comment:
constructor () {
    this.vikingArmy = [];
    this.saxonArmy = [];
  }

  addViking (viking) {
    this.vikingArmy.push(viking);
  }

  addSaxon (saxon) {
    this.saxonArmy.push(saxon);
  }
// Jose C. approach:
  _genericAttack (firstArmy, secondArmy) {
    const firstCount = firstArmy.length;
    const secondCount = secondArmy.length;
    const firstIndex = randomInteger(0, firstCount - 1);
    const secondIndex = randomInteger(0, secondCount - 1);

    const first = firstArmy[firstIndex];
    const second = secondArmy[secondIndex];
    
    const firstStrength = first.strength;
    const attackMessage = second.receiveDamage(firstStrength);

    for (let second of secondArmy) {
      if (second.health <= 0) {
        const index = secondArmy.indexOf(second);
        secondArmy.splice(index, 1);
      }
    }

    return attackMessage;
  }

  vikingAttack () {
    return this._genericAttack(this.vikingArmy, this.saxonArmy);
  }

  saxonAttack () {
    return this._genericAttack(this.saxonArmy, this.vikingArmy);
  }

  showStatus () {
    if (this.saxonArmy.length === 0) 
        return "Vikings have won the war of the century!";
    else if (this.vikingArmy.length === 0) 
        return "Saxons have fought for their lives and survive another day...";
    else if (this.saxonArmy.length >= 1 && this.vikingArmy.length >= 1) 
        return "Vikings and Saxons are still in the thick of battle.";
  }